src/dotnet/CommandFactory/CommandResolution/LocalToolsCommandResolver.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.DotNet.ToolManifest;
+using Microsoft.DotNet.ToolPackage;
+using NuGet.Frameworks;
+using Microsoft.DotNet.Cli;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.CommandFactory
+{
+    internal class LocalToolsCommandResolver : ICommandResolver
+    {
+        private readonly ToolManifestFinder _toolManifest;
+        private readonly ILocalToolsResolverCache _localToolsResolverCache;
+        private readonly IFileSystem _fileSystem;
+        private readonly DirectoryPath _nugetGlobalPackagesFolder;
+        private const string LeadingDotnetPrefix = "dotnet-";
+
+        public LocalToolsCommandResolver(ToolManifestFinder toolManifest = null,
```

Comment:
the nittiest of nits: for multiline parameter lists, I think the first parameter should also go with the other parameters, indentation-wise (i.e. start a new line before the first one).

Reply:
I don't, but if your editor is doing this, then I'm fine with resolving this conversation without changing.